tests/components/zha/test_sensor.py

```diff
@@ -169,7 +169,7 @@ async def async_test_smart_energy_summation(hass, cluster, entity_id):
     assert hass.states.get(entity_id).attributes["device_type"] == "Electric Metering"
     assert (
         hass.states.get(entity_id).attributes[ATTR_DEVICE_CLASS]
-        == SensorDeviceClass.ENERGY
```

Comment:
Was this change intended? Should the device class be volume for an electric metering entity? I’ll take a closer look at the PR as a whole when I have a few but this stuck out when peeking quickly from my phone.

Reply:
Yes, this was intended, but I was also confused, as the line above says it has "m³" as UoM.

```python
assert_state(hass, entity_id, "12.32", UnitOfVolume.CUBIC_METERS)
```

Reply:
I didn’t scroll past what was immediately visible in the diff. I’ll look closer when I’m at a computer.

Reply:
I can't comment on the line itself since it wasn't changed as part of this... Please update the data on line 338:

```
(
            smartenergy.Metering.cluster_id,
            "summation_delivered",
            async_test_smart_energy_summation,
            9,
            {
                "demand_formatting": 0xF9,
                "divisor": 1000,
                "metering_device_type": 0x00,
                "multiplier": 1,
                "status": 0x00,
                "summation_formatting": 0b1_0111_010,
                "unit_of_measure": 0x01,
            },
            {"instaneneous_demand"},
        ),
```

so that this test uses data all from one sensor type (so we don't have mixed energy and volume stuff)

Reply:
Done, the state test has a decimal point more as before, as the if in formatter on line 695 is now False.